w_flux_codemod/README.md

```diff
@@ -0,0 +1,89 @@
+# w_flux_codemod
+
+> **Built with [dart_codemod][dart_codemod].**
+
+A codemod to convert existing usages of non null-safe `Action` to null-safe `ActionV2`.
+
+## Motivation
+
+`w_flux` was upgraded to dart 3 and made null safe, but we ran into an issue when migrating the `Action` class.
+
+The `Action` class has a call method with an optional `payload` paramater that now must be typed as nullable. However, this means that we cannot make `listener` payloads non-nullable, since there's no guarantee that the argument was specified.
+
+```
+class Action<T> /*...*/ {
+  Future call([T? payload]) { 
+    for (final listener in _listeners) {
+      await listener(payload);
+      //             ^^^^^^^
+      // Error: can't assign T? to T
+    }
+  }
+  
+  ActionSubscription listen(dynamic onData(T event)) {/*...*/}
+  
+  /*...*/
+}
+```
+
+To be able to support non-nullable payloads (in addition to nullable payloads), we made a new `ActionV2` class with required payloads.
+
+## Usage
+
+1. Ensure you have the codemod package installed.
+    ```bash
+    dart pub global activate w_flux_codemod
+    ```
+
+2. Run the codemod:
+
+    - step by step:
+    ```bash
+    dart pub global run w_flux_codemod:action_v2_migrate_step_1.dart
+    dart pub global run w_flux_codemod:action_v2_migrate_step_2.dart
+    ```
+
+    - all at once:
+    ```bash
+    dart pub global run w_flux_codemod:action_v2_migrate.dart
```

Comment:
Shouldn't need the `.dart` here since we're defining names for these executables in the pubspec:
```suggestion
    dart pub global run w_flux_codemod:action_v2_migrate_step_1
    dart pub global run w_flux_codemod:action_v2_migrate_step_2
    ```

    - all at once:
    ```bash
    dart pub global run w_flux_codemod:action_v2_migrate
```